### ISSUE TITLE
Rename rather than copy if tmp2 filename is on the same device as final filename

### ISSUE THREAD

**Issue by cross** [Chia-Network/chiapos#313]
Drawing this over from Chia-Network/chia-blockchain#8847 -

When tmp and final files are in different directories on the same device, the code doesn't know that it can perform a rename(), and instead spends a lot of time unnecessarily copying the file data.

**Comment by cross** [comment]
Not stale, waiting on #314

**Comment by cross** [comment]
Not stale, waiting on #314

**Comment by cross** [comment]
This is an ongoing irritation in my environment.  Can I help in any way getting #314 testing completed so it can be merged?

**Comment by cross** [comment]
Not stale, still need #314 merged.

**Comment by cross** [comment]
Still waiting.  @hoffmang9 , can someone get this in please?

**Comment by wjblanke** [comment]
We'll put this into 1.3 beta so we can get a better idea of how it works in practice with different drives, devices, os, mount point, etc. Need to do another chiapos release then merge into main
